docs/concepts/internationalization.md

```diff
@@ -0,0 +1,57 @@
+
+Localization will happen in the game's locale. In a Minecraft client the locale is specified by the language settings. On a dedicated server, the only supported locale is en_US.
+
+Language files
+--------------
+
+Language files are located by `assets/domain/lang/locale.lang` (e.g. the US English translation for `examplemod` would be `assets/examplemod/lang/en_us.lang`). Resource pack format 3 requires the locale name to be lowercased. The file format is simply lines of key-value pairs. Lines starting with `#` are treated as comments.
+
+```properties
+# items
+item.examplemod.example_item.name=Example Item Name
+
+# blocks
+tile.examplemod.example_block.name=Example Block Name
+
+# commands
+commands.examplemod.examplecommand.usage=/example <value>
+```
+
+A language file may contain escape sequences if the file starts with the comment `#PARSE_ESCAPES`.
+
+Localization methods
+--------------------
+
+!!! warning
+    A common issue is having the server localize for clients. The server can only localize in its own locale, which does not necessarily match the locale of connected clients.
+    
+    To respect the language settings of clients, the server should have clients localize text in their own locale using TextComponentTranslation or other methods preserving the language neutral translation keys.
+
+### `net.minecraft.client.resources.I18n` (client only)
+
+**This I18n class can only be found on a Minecraft client!** It is intended to be used by code that only runs on the client. Attempts to use this on a server will throw exceptions and crash.
+
+- `format(String, Object...)` localizes in the client's locale with formatting. The first parameter is a language key, and the rest are used for `String.format(String, Object...)`.
+
+### `net.minecraft.util.text.translation.I18n` (deprecated)
+
+**This class is deprecated and should often be avoided.** It is intended to be used by code that runs on the logical server. Because localization should rarely happen on the server, other alternatives should be considered.
+
+- `translateToLocal(String)` localizes in the game's locale without formatting. The parameter is a language key.
+- `translateToLocalFormatted(String, Object...)` localizes in the game's locale with formatting. The first parameter is a language key, and the rest are used for `String.format(String, Object...)`.
+
+### `TextComponentTranslation`
+
+`TextComponentTranslation` is an `ITextComponent` that is localized and formatted lazily. It is very useful when sending messages to players because it will be automatically localized in their own locale.
+
+The first parameter of the `TextComponentTranslation(String, Object...)` constructor is a language key, and the rest are used for formatting. The only supported format specifiers are `%s` or `%1$s`. Formatting arguments may be other `ITextComponent`s that will be inserted into the resulting formatted text with all their attributes preserved.
```

Comment:
maybe specify that the positional arg can be any number? e.g. `%2$s`, `%3$s`, `%<n>$s`
In case people weren't familiar with that notation for String.format (I wasn't until just now)